apps/aeutils/src/aeu_mp_trees.erl

```diff
@@ -0,0 +1,543 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2017, Aeternity Anstalt
+%%% @doc Merkle Patricia Trees
+%%%
+%%% @end
+%%%-------------------------------------------------------------------
+
+-module(aeu_mp_trees).
+
+-export([ new/0
+        , new/2
+        , delete/2
+        , get/2
+        , pp/1
+        , put/3
+        , root_hash/1
+        ]).
+
+-export_type([tree/0
+             ]).
+
+-record(mpt, { hash = <<>> :: <<>> | hash() %% <<>> for the empty tree
```

Comment:
In the spirit of using one specific hash function, would it make sense to use the type of that hash library, e.g. aec_sha256:hash()?

Reply:
Agreed. There is a ticket for doing that, and the current available abstracted hash function is specifying the sha256, so I didn't bother using it.